src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,209 @@
+                    LOG.warn(message);
+                    NotifyUsersForSubscriptionJob.createNotification(
+                        "project-updated",
+                        project.id,
+                        message
+                    );
+                    continue;
+                }
+                // Send deployment (with new feed version) to most recently used server.
+                String latestServerId = deployment.latest().serverId;
+                OtpServer server = Persistence.servers.getById(latestServerId);
+                if (server != null) {
+                    // Advance all feed versions in the deployment.
+                    deployment.feedVersionIds = new ArrayList<>();
+                    deployment.feedVersionIds.addAll(feedVersionIdsForDeployment);
+                    Persistence.deployments.replace(deployment.id, deployment);
+                    // Queue up the deploy job.
+                    if (DeploymentController.queueDeployJob(new DeployJob(deployment, owner, server))) {
+                        LOG.debug("Last auto deploy date updated for project {}.", project.name);
+                        project.lastAutoDeploy = new Date();
+                        Persistence.projects.replace(project.id, project);
+                    } else {
+                        LOG.debug("Could not auto-deploy to {} due to conflicting active deployment for project {}.",
+                            server.name,
+                            project.name);
+                    }
+                } else {
+                    LOG.debug("Server with id {} no longer exists. Skipping deployment for project {}.",
+                        latestServerId,
+                        project.name);
+                }
+            } finally {
+                lockedProjects.remove(project.id);
+                LOG.debug("Auto deploy lock removed for project id: {}", project.id);
+            }
+        }
+        status.completeSuccessfully("Auto deploy complete.");
+    }
+
+    /**
+     * Retrieve feed versions from all feed sources which have no critical errors and were created after the last
+     * auto deploy.
+     */
+    private List<String> getFeedVersionsForDeployment(Project project, Deployment deployment) {
+        List<String> feedVersionIds = new ArrayList<>();
+        Collection<FeedSource> feedSources = getDeploymentFeedSources(deployment);
+        for (FeedSource feedSource : feedSources) {
+            for (FeedVersion feedVersion : feedSource.retrieveFeedVersions()) {
+                if ((project.lastAutoDeploy == null ||
+                    feedVersion.dateCreated.after(project.lastAutoDeploy)) &&
+                    !feedVersion.hasCriticalErrors()) {
+                    // Feed version has no critical errors, advance the feed version in the deployment.
+                    feedVersionIds.add(feedVersion.id);
+                }
+            }
+        }
+        return feedVersionIds;
+    }
```

Comment:
This isn't quite the right logic. As this is currently written, only those feed versions that were created after the last AutoDeployJob and without critical errors get added to this updated list. That would result in some feed sources being removed from the deployment. There could be a number of feed sources where their feed version hasn't been updated since the last project auto deployment occurred. Some agencies update their feeds very frequently, some don't. 

I've gone ahead and made PR #372 which has the logic which I think should work.

Reply:
Addressed in Evan's PR.